packages/components/src/popover/index.js

```diff
@@ -118,6 +118,9 @@ const placementToAnimationOrigin = ( placement ) => {
 	return x + ' ' + y;
 };
 
+const isTopBottomPlacement = ( placement ) => /top|bottom/i.test( placement );
+const hasBeforePlacement = ( placement ) => /top|left/i.test( placement );
```

Comment:
We may even change this regex to force the top/bottom/left to be at the start of the string?

Reply:
Yes, a `startsWith` could be enough I think..

Reply:
I had initially used a RegEx for its compact form, but I switched to `startsWith` as you suggested in https://github.com/WordPress/gutenberg/pull/42950/commits/43c1d8db0013d15e85c51e8ceeda7132639e06e7